orttraining/orttraining/python/training/orttrainer_options.py

```diff
@@ -662,6 +677,15 @@ def _check_is_callable(field, value, error):
                 'type': 'integer',
                 'min': 0,
                 'default': 0
+            },
+            'propagate_cast_ops_level': {
+                'type': 'integer',
+                'default': -1
```

Comment:
what is the min value? what is the max?

Reply:
The min value is -1 and the max value can be anything. Right now we have implementation for 0, 1 and 2. Anything above 2 will be same as 2. In future we might add new optimizations at higher levels.